lib/pbench/server/api/resources/__init__.py

```diff
@@ -631,9 +637,13 @@ def _check_authorization(self, user: Union[str, None], access: Union[str, None])
         If there is no current authenticated API user, only READ operations on
         public data will be allowed.
 
-        If the specified "user" context of the operation is None, we default to
-        the authenticated caller, if any; Pbench query operations will do the
-        same.
+        If the specified "user" context of the operation is None, the context
+        is defaulted based on the authenticated user: for an ADMIN user, this
+        means implicitly "all users" (to which the ADMIN has unlimited access);
+        for an unauthenticated user, no access to another "user" is allowed at
+        all, and will be caught at convert_username(); for an authenticated
```

Comment:
Unauthenticated clients can see public data, but not by username. So `{}` shows them all public data, `{"access": "public"}` shows them all public data, but they can't probe for valid users with `{"user": "drb", "access": "public"}` (which would work if `drb` exists, possibly with an empty return if "drb" has none, and would fail otherwise).

Basically, he idea was not to allow an unauthenticated client to ask "does this username exist"?

And, yeah, the other option is to allow `{"user": "drb", "access": "public"}` and treat usernames as public. I considered this; and I'm not entirely consistent on this. For example, the `datasets/list` query can ask for `"metadata": ["dataset.owner"]` currently without restriction, so we'll reveal the owner of a public dataset. Arguably, if we can't ask for `{"user": "drb", "access": "public"}` we shouldn't reveal `dataset.owner` on unauthenticated queries, either.

I really don't like any of the alternatives, but this approach seemed to make more sense than the others. So far...